### ISSUE TITLE
Trying to train Terramind

### ISSUE THREAD

**Issue by ArthurDelannoyazerty** [IBM/terramind#4]
Hello, 
I have a project where I need to train that model. I am quite new to the terratorch library, but as I understand it, there is no easy way to train Terramind. 
I tried to use the ReconstructionTask from TerraTorch (which seems to be the right task because Terramind reconstruct the input and do a cross entropy between the input and ouput) but the ReconstructionTask doe not provide a cross entropy loss. I understood that the ReconstructionTask only work for continuous variable, which is not the case here because TerraMind use discrete token reconstruction.

Can you make the training code available ? Or give some explanation on how you did this (did you use TerraTorch for the training ?)

Is it easier to use TerraTorch or the raw model for training ?

Also, I tried to use the "caption" and "coord" modality, but that gave me a `NotImplementedError`. What text tokenizer did you used ? For the "coord" text, how did you created the 2 special tokens ?

**Comment by blumenstiel** [comment]
Hi @ArthurDelannoyazerty, 
do you want to do continous pre-training or fine-tuning?

The `ReconstructionTask` only works with MAE models like Prithvi but not with TerraMind, which is token based.

The open-sourced model is meant to be used as zero-shot or with fine-tuning, e.g. using the `SemanticSegmentationTask` from TerraTorch. We do not open-soure the pre-training code as it is quite complicated and not very feasible for users to do (the code is adapted on 4M which is open-source).

About the captions and coords. This is still a todo for me. However, I don't think I have the time to implement it within the next 4 weeks. The open-sourced models only support coords, captions are not yet incuded. We use a simple lookup tokenizer for it that include the lat lon tokens.

**Comment by ArthurDelannoyazerty** [comment]
Hi, 
I am trying to pre-train the model with additionnal modalities and data.
I was not aware of the 4M framework. I am looking into it and it is very interesting, thank you for the info.
I will try to use 4M for my custom pre-training, and create my own tokenizer for my data.

Thank you for your reponse
